core/app/components/PathAwareDialogButton.tsx

```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import type { ButtonProps } from "ui/button";
+import { Button } from "ui/button";
+import { Loader2 } from "ui/icon";
+import { cn } from "utils";
+
+import { usePathAwareDialogSearchParam } from "~/lib/client/usePathAwareDialogSearchParam";
+
+export type PathAwareDialogueProps = {
+	variant?: ButtonProps["variant"];
+	size?: ButtonProps["size"];
+	className?: string;
+	children?: React.ReactNode;
+	id: string;
+};
+
+export const PathAwareDialogButton = (props: PathAwareDialogueProps) => {
+	const { isPending, toggleDialog: toggleDialogue } = usePathAwareDialogSearchParam({
+		id: props.id,
+	});
+	// Should maybe be a link, but then it plays less nicely with `nuqs`
```

Comment:
this is less of a "best practices" question and more a philosophical question. Is this a button that open a dialogue that happens to use a queryparam to keep state, or is it a link to a page that happens to open a dialog that covers the current page?

you could argue both, but i think having it be a button is fine. if we wanted it to be a link, we would have to manually construct the url ourselves again (find current url, construct the correct query param, add it to the already existing query params etc), which negates some of the benefits of using `nuqs` (it does all of that for us)